staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -43,7 +43,14 @@ import (
 	"k8s.io/utils/trace"
 )
 
-const defaultExpectedTypeName = "<unspecified>"
+const (
+	defaultExpectedTypeName = "<unspecified>"
+
+	// ReflectorTypeAnnotationKey is the name of the annotation key used to indicate the expected type
+	// in an exampleType passed to NewReflector, where the exampleType does not have its TypeMeta
+	// specified (for example, unstuctured.Unstructured).
+	ReflectorTypeAnnotationKey = "reflector.cache.k8s.io/type"
```

Comment:
I think "type" is misleading here, "resource" or "GVR" would be better.

Reply:
I originally had this as gvr, but waffled and made it more generic. I can do resource or gvr if you'd prefer.